### PR TITLE
Remove old options and replace with new global ones for string search

### PR DESCRIPTION
**Pull request by wargio**
# do not squash

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This PR allows to customize the options for searching strings and simplifies how RzBin and RzBinPlugins finds strings.

### CODE REVIEW

File: test/db/analysis/arm [hunk 1]
```diff
@@ -1067,13 +1067,13 @@ RUN
 NAME=misaligned arm string xref
 FILE=bins/mach0/misaligned_data-iOS-armv7
 CMDS=<<EOF
-aav
-aae
-af
-axt @ str.helloradareworld
+aaa
+axt @ str.helloradareworl
```
Comment (XVilka) [inline test/db/analysis/arm:1071]: Any clue why it now cuts more than before?

File: test/db/esil/apple [hunk 2]
```diff
@@ -86,7 +86,7 @@ EXPECT=<<EOF
 0x00002cc0 bl sym.imp.objc_msgSend
 0x00002cd0 str.objectForKey:
 0x00002cec bl sym.imp.objc_msgSend
-0x00002cfc str.cstr.file_name_is:_____protection_class:
+0x00002cfc str.cstr.file_name_is:_____protection_class
```
Comment (XVilka) [inline test/db/esil/apple:89]: Same here

**Review by XVilka — APPROVED** [review]
Apart from those 2 questions - LGTM

File: test/db/analysis/arm (hunk 1 above)
Comment (wargio) [reply test/db/analysis/arm:1071]: no idea, i think is the filter doing stuff.

**Pull request merged**